pkg/microservice/reaper/core/service/reaper/reaper.go

```diff
@@ -429,6 +430,45 @@ func (r *Reaper) AfterExec() error {
 		}
 	}
 
+	if r.Ctx.UploadEnabled {
+		forcedPathStyle := true
+		if r.Ctx.UploadStorageInfo.Provider == setting.ProviderSourceAli {
+			forcedPathStyle = false
+		}
+		client, err := s3.NewClient(r.Ctx.UploadStorageInfo.Endpoint, r.Ctx.UploadStorageInfo.AK, r.Ctx.UploadStorageInfo.SK, r.Ctx.UploadStorageInfo.Insecure, forcedPathStyle)
+		if err != nil {
+			return fmt.Errorf("failed to create s3 client to upload file, err: %s", err)
+		}
+		for _, upload := range r.Ctx.UploadInfo {
+			info, err := os.Stat(upload.FilePath)
+			if err != nil {
+				return fmt.Errorf("failed to upload file path [%s] to destination [%s], the error is: %s", upload.FilePath, upload.DestinationPath, err)
+			}
+			// if the given path is a directory
+			if info.IsDir() {
```

Comment:
If it is a directory, can it be compressed and then uploaded?

Reply:
Not the intended way